org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/codeActions/ICodeActionService.xtend

```diff
@@ -12,15 +12,38 @@ import org.eclipse.lsp4j.CodeAction
 import org.eclipse.lsp4j.CodeActionParams
 import org.eclipse.lsp4j.Command
 import org.eclipse.lsp4j.jsonrpc.messages.Either
+import org.eclipse.xtend.lib.annotations.Accessors
 import org.eclipse.xtext.ide.server.Document
+import org.eclipse.xtext.ide.server.ILanguageServerAccess
 import org.eclipse.xtext.resource.XtextResource
 import org.eclipse.xtext.util.CancelIndicator
 
 /**
  * @author Sven Efftinge - Initial contribution and API
+ * @deprecated use {@link ICodeActionService2} instead
  */
+@Deprecated
 interface ICodeActionService {
 	
 	def List<Either<Command, CodeAction>> getCodeActions(Document document, XtextResource resource, CodeActionParams params, CancelIndicator indicator)
+		
+}
+
+/**
+ * @author Jan Koehnlein
+ * @since 2.18
+ */
+ interface ICodeActionService2 {
 	
-}
\ No newline at end of file
+	def List<Either<Command, CodeAction>> getCodeActions(Options options)
+		
+	@Accessors
```

Comment:
Do we want to make this mutable, or did you use `@Accessor` instead of `@Data` for the easier instantiation in Xtend (`new Option() => []`)?

Reply:
The latter. These are quite a lot of params, a bit to many for a constructor IMHO.

Reply:
OK, thanks. I am OK with both, just wanted to understand your intention.